openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/FairGBMBinaryClassificationModelTrainerTest.java

```diff
@@ -0,0 +1,270 @@
+        );
+
+        final ArrayList<List<Double>> scoresWithSingleChunk = fitModelAndGetFirstScoresPerClass(
+                DATASET_RESOURCE_NAME,
+                TestSchemas.CATEGORICALS_SCHEMA_LABEL_AT_START,
+                MAX_NUMBER_OF_INSTANCES_TO_TRAIN,
+                MAX_NUMBER_OF_INSTANCES_TO_SCORE,
+                10000000 // Try to have a single chunk (10M instances / chunk)
+        );
+
+        assertThat(scoresWithSmallChunks).as("scores")
+                                         .isEqualTo(scoresWithChunks)
+                                         .isEqualTo(scoresWithSingleChunk);
+    }
+
+    /**
+     * Assert that there's an error when training with no instances.
+     */
+    @Test
+    public void fitWithNoInstances() {
+
+        final List<Instance> noInstances = new ArrayList<>();
+        final Dataset emptyDataset = new MockDataset(TestSchemas.CATEGORICALS_SCHEMA_LABEL_AT_START, noInstances);
+
+        assertThatThrownBy(() ->
+                new LightGBMModelCreator().fit(
+                        emptyDataset,
+                    new Random(),
+                    MODEL_PARAMS
+                )
+        )
+        .isInstanceOf(RuntimeException.class);
+    }
+
+    /**
+     * Test Feature Contributions with target at end.
+     *
+     * @throws URISyntaxException For errors when loading the dataset resource.
+     * @throws IOException        For errors when reading the dataset.
+     * @since 1.3.0
+     */
+    @Test
+    public void testFeatureContributionsTargetEnd() throws URISyntaxException, IOException {
+        final Dataset dataset = CSVUtils.getDatasetWithSchema(
+                TestResources.getResourcePath(DATASET_RESOURCE_NAME),
+                TestSchemas.CATEGORICALS_SCHEMA_LABEL_AT_END,
+                10000
+        );
+        ensureFeatureContributions(dataset, MODEL_PARAMS);
+    }
+
+    /**
+     * Test Feature Contributions with target at middle.
+     *
+     * @throws URISyntaxException For errors when loading the dataset resource.
+     * @throws IOException        For errors when reading the dataset.
+     * @since 1.3.0
+     */
+    @Test
+    public void testFeatureContributionsTargetMiddle() throws URISyntaxException, IOException {
+        final Dataset dataset = CSVUtils.getDatasetWithSchema(
+                TestResources.getResourcePath(DATASET_RESOURCE_NAME),
+                TestSchemas.CATEGORICALS_SCHEMA_LABEL_IN_MIDDLE,
+                10000
+        );
+        ensureFeatureContributions(dataset, MODEL_PARAMS);
+    }
+
+    /**
+     * Test Feature Contributions with target at beginning.
+     *
+     * @throws URISyntaxException For errors when loading the dataset resource.
+     * @throws IOException        For errors when reading the dataset.
+     * @since 1.3.0
+     */
+    @Test
+    public void testFeatureContributionsTargetBeginning() throws URISyntaxException, IOException {
+        final Dataset dataset = CSVUtils.getDatasetWithSchema(
```

Comment:
Ensure that the file loaded is properly closed

Reply:
Reading the dataset is done via a `CSVParser` which AFAIK closes automatically when it falls out of scope.

Nonetheless I added a `csvParser.close()` to `CSVUtils::getDatasetWithSchema`